lib/Basic/SourceLoc.cpp

```diff
@@ -24,7 +24,11 @@ void SourceManager::verifyAllBuffers() const {
   };
 
   // FIXME: This depends on the buffer IDs chosen by llvm::SourceMgr.
-  __attribute__((used)) static char arbitraryTotal = 0;
+  // _attribute__((used) is not defined in MSVC
+#if !defined(_MSC_VER)
+  __attribute__((used))
+#endif
```

Comment:
Hmm, I wonder if there is a better way to handle this.  Something like using a `__pragma(warning(supress:????))))`.  In that case, we could have a `SWIFT_USED` macro.